map/map-test.js

```diff
@@ -1044,3 +1042,39 @@ QUnit.test('Deep updating a map', function() {
 	QUnit.equal(obj.list[0], 'something', 'the first element of the list should be updated');
 
 });
+
+testHelpers.dev.devOnlyTest("Error on not using a constructor or string on short-hand definitions (#278)", function() {
+	expect(1);
+	var message = "A Constructor or string must be used with shorthand definitions: https://canjs.com/doc/can-define.types.propDefinition.html#String";
+	var finishErrorCheck = testHelpers.dev.willError(message);
+
+	// Having JSHint ignore this because we don't need to actually
+	// use 'VM' anywhere to get the errors to show and the test to complete
+	/* jshint ignore:start */
+	var VM = DefineMap.extend({
+		prop01: 0,
+		prop02: function() {},
+		prop03: 'string',
+		prop04: DefineMap,
+		prop05: "a string that is not a type",
+		prop06: [],
+		get prop07() {},
+		set prop07(newVal) {},
+		prop08: 'boolean'
+	});
+	/* jshint ignore:end */
+
+	QUnit.equal(finishErrorCheck(), 2);
```

Comment:
Maybe each of these cases needs to be tested separately? This doesn't actually test which two cause the warning. If you add the `prop` to the warning message you could also make this use a regex to check that the two correct values show the warning.

Something like
```
message = /badProp.*does not match any of the supported definitions/;

...

DefineMap.extend({
    badProp01: 0,
    prop02: function() {},
    prop03: 'string',
    prop04: DefineMap,
    badProp05: "a string that is not a type",
    prop06: [],
    get prop07() {},
    set prop07(newVal) {},
    prop08: 'boolean'
});
```